src/extension/viewed-state.ts

```diff
@@ -0,0 +1,44 @@
+import * as vscode from "vscode";
```

Comment:
The file uses the extension context which is not available in a webview, that's why I put it in `src/extension/`. The type `ViewedState` is used in two files in `src/webview/` but both already import another type from `src/extension/` so I thought it's OK to keep the type in this file, too, even though it could be seen as shared. 

There's the `src/shared/message.ts` file which might contain shared types, except that it would lead to circular dependencies with message API definitions.

Maybe there are some other types that, together with `ViewedState`, could live in a new file like `src/shared/types.ts`?

Reply:
I see! Yeap, I think we are good to go the way you did. Let's keep it that way. :)